samples/bluetooth/fast_pair/locator_tag/src/fp_adv.c

```diff
@@ -405,14 +401,43 @@ BT_CONN_CB_DEFINE(conn_callbacks) = {
 
 static void fp_adv_provisioning_state_changed(bool provisioned)
 {
+	struct bt_le_oob oob;
+	int err;
+
 	fmdn_provisioned = provisioned;
+
+	if (!provisioned) {
+		/* Force the RPA rotation to synchronize the Fast Pair advertising
+		 * payload with its RPA address using rpa_expired callback.
+		 */
+		err = bt_le_oob_get_local(fp_adv_param.id, &oob);
+		if (err) {
+			LOG_ERR("Fast Pair: bt_le_oob_get_local failed: %d", err);
+		}
+	}
 }
 
 static struct bt_fast_pair_fmdn_info_cb fmdn_info_cb = {
 	.provisioning_state_changed = fp_adv_provisioning_state_changed,
 };
 
-void app_fp_adv_mode_set(enum app_fp_adv_mode adv_mode)
+enum app_fp_adv_mode app_fp_adv_mode_get(void)
+{
+	return fp_adv_mode;
+}
+
+static bool fp_adv_trigger_is_any_enabled(void)
+{
+	STRUCT_SECTION_FOREACH(app_fp_adv_trigger, adv_trigger) {
+		if (adv_trigger->enabled) {
+			return true;
+		}
+	}
+
+	return false;
+}
+
+static void fp_adv_mode_set(enum app_fp_adv_mode adv_mode)
```

Comment:
`fp_advertising_start` is quite a confusing function name here, because under the hood it can actually be used to stop advertising

Reply:
Maybe we could name it `ble_stack_update`/`ble_stack_notify` (or we could flatten function calls here)?

Reply:
Improving naming could significantly simplify understanding the code here

Reply:
Agreed but I would rather not to refactor whole module if not needed as I am not touching directly those function.
Discussed to rename `fp_advertising_start` -> `fp_advertising_update` and `fp_adv_start_internal` -> `bt_stack_advertising_update`